File: String/SpaceReplace/SpaceReplace.java
class SpaceReplace {
	public char[] replaceString(char[] str, int len) {
		if(len == 0 || str == null)
			return null;
		
		
		int spaces = 0;
		for(char c : str) {
			if (c == ' ') {
				space++;
			}
		}
		
		int index = len + spaces * 2 - 1;
		for(int i = len - 1; i >  0; i--) {
			if(str[i] != ' ') {
				str[index] = str[i];
				index--;
			} else {
				str[index--] = '0';
				str[index--] = '2';
				str[index--] = '%';
			}
		}
		return str;
	}
}

File: String/SpaceReplace/SpaceReplace.py
class SpaceReplace:
    def spaceReplace(self, str, len):
        if str is none:
            return len
            
            
        spaces = 0
        for c in str:
            if c == ' ':
                spaces += 1
                
                
        newLen = len + spaces * 2
        index = newLen - 1
        for i in range(len - 1, -1, -1):
            if str[i] == ' ':
                str[index] = '0'
                str[index - 1] = '2'
                str[index - 2] = '%'
                index -= 3
            else:
                str[index] = str[i]
                index -= 1
                
         return str
            
        
    
        